### ISSUE TITLE
Inconsistency between RenderLink and RenderImage parameters

### ISSUE THREAD

**Issue by Gatogordo** [mikeedwards83/Glass.Mapper#203]
Hi,

The RenderLink function in GlassHtml has the ability to work with a NameValueCollection as parameters which makes it easy to use generic function to create these parameter objects.

The RenderImage function however does not have this.. Is it possbile to have that there as well?

Thx,
Gert

**Comment by Andrew-Oliveira** [comment]
This should not be an **enhancement**, this is more of a **bug** or at the very least it is **laziness**. There's a certain amount of assumption that comes with consuming the code of others. Having similarly named methods that take similar, (and vague), parameters is awful. It's easy to assume they behave similarly. Plus the only thing distinguishing the parameter in question is the name. (RenderLink has it named "attributes", RenderImage has it named "parameters".) Not to mention that you're limiting the usefulness of the code by forcing people to use an anonymous object. (WTF?!)

Adding an "attributes" parameter and accompanying logic can't be that difficult, especially when it exists elsewhere. If there's concern about potential confusion between "attributes" and "parameters", I'd suggest something specific like "scImageParameters" or something.

Sorry if I seem a little annoyed, but I just spent hours digging only to hit this wall. Please fix this!

**Comment by cardinal252** [comment]
I disagree that an API inconsistency is a bug. This can be easily overcome by implementing your own extended GlassHtml and registering it at startup. Mappers can also be overridden  Therefore, this is not a 'wall' either in my opinion. It is an inconsistency in an API of a product built over many years by folk in there spare time.

Needless to say, we will endeavour to include it when it can be addressed

@Andrew-Oliveira I would remind you that we offer a paid consultancy if this needs to be addressed as a priority. Otherwise, this is an open source project ran for free by a limited number of people in their spare time. As an open source project, you may submit a Pull Request for consideration.
